Week3/homework/step2-7.js

```diff
@@ -20,4 +20,20 @@ f2(y);
 
 console.log(y);
 
-// Add your explanation as a comment here
+/* With the first console log, the value of the x will be printed as 9. Firstly, the value of x 
+is set to 9 which is a number and thus primitive data type. Primitive data types cannot be 
+reassigned when defined as const which is the case here. Even if x is defined as 'let', the value of 
+it would still be 9 since the value of x is passed into the function f1 and x is not assigned to a 
+new value. The value of x would have changed if we wrote it like this: 
```

Comment:
What you have written for primitive is correct but the actual reason why value of `x` doesn't change after the function is because JavaScript as a language is call by value. This means that the value of `x` is passed to the function and not the variable itself. Which is why modifying the value inside the function wouldn't have an impact on the variable which was passed as argument.

As far as objects in JavaScript is concerned, what is passed to the function is not the value but the reference of the object. Thus, if we modify it inside the function, it modifies the variable itself which is also pointing to the same reference. Hope it makes sense. 🙂